x/leverage/keeper/msg_server.go

```diff
@@ -31,21 +31,24 @@ func (s msgServer) Supply(
 		return nil, err
 	}
 
-	if err := s.keeper.Supply(ctx, supplierAddr, msg.Asset); err != nil {
+	received, err := s.keeper.Supply(ctx, supplierAddr, msg.Asset)
+	if err != nil {
 		return nil, err
 	}
 
 	s.keeper.Logger(ctx).Debug(
 		"assets supplied",
 		"supplier", supplierAddr.String(),
-		"amount", msg.Asset.String(),
+		"supplied", msg.Asset.String(),
+		"received", received.String(),
 	)
 
 	ctx.EventManager().EmitEvents(sdk.Events{
 		sdk.NewEvent(
 			types.EventTypeSupply,
 			sdk.NewAttribute(types.EventAttrSupplier, supplierAddr.String()),
-			sdk.NewAttribute(sdk.AttributeKeyAmount, msg.Asset.String()),
+			sdk.NewAttribute(types.EventAttrSupplied, msg.Asset.String()),
+			sdk.NewAttribute(types.EventAttrReceived, received.String()),
```

Comment:
side note: with new features in 0.46 (responses registered on chain), the events have less sense.